### PR TITLE
Support a configurable log format

### PR DESCRIPTION
**Pull request by esmet**
## Description
Allows an operator to set the text log format. Useful for creating custom logs, since the ones out of the box may or may not contain all of the useful information for someone's particular use case.

## Testing
Tested by hand

## Todos
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: ambassador/ambassador/envoy/v2/v2listener.py [hunk 1]
```diff
@@ -535,11 +535,15 @@ def __init__(self, config: 'V2Config', listener: IRListener) -> None:
                 } ]
             else:
                 # Use a sane access log spec
+                log_format = config.ir.ambassador_module.get('envoy_log_format', None)
+                if not log_format:
+                    log_format = 'ACCESS [%START_TIME%] \"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%\" %RESPONSE_CODE% %RESPONSE_FLAGS% %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% \"%REQ(X-FORWARDED-FOR)%\" \"%REQ(USER-AGENT)%\" \"%REQ(X-REQUEST-ID)%\" \"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\"'
+                config.ir.logger.info("V2Listener: Using log_format '%s'" % log_format)
```
Comment (esmet) [inline ambassador/ambassador/envoy/v2/v2listener.py]: I don't have strong opinions on whether this log line deserves to stay or not. Let me know.

Comment (kflynn) [reply ambassador/ambassador/envoy/v2/v2listener.py]: I think it should probably be `debug` rather than `info`. The logs are pretty cluttered as is. 🙁

**Comment by inercia** [comment]
There are a couple of conflicts, so @esmet if you could rebase ad resolve those conflicts we could merge this PR, and then I will add a tests in a subsequent PR...

**Comment by inercia** [comment]
Closed in favor of #1969

**Pull request closed without merging**